bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -93,6 +95,7 @@ static PendingAddOp create(LedgerHandle lh, ByteBuf payload, AddCallback cb, Obj
         op.callbackTriggered = false;
         op.hasRun = false;
         op.requestTimeNanos = Long.MAX_VALUE;
+        op.allowFastFailOnBlockedWrite = false;
```

Comment:
Do we really need this mechanism to pass the fastFail? Why not make it default if throttle time is enabled? or come up with another configuration option? Here we are doing it by default in the read and write path, and for sure LaC we can do the same.

Reply:
This is unrelated to client configuration and enabled in runtime per op. 
This is a flag on the op which controls the behavior in case if netty channel is not writable:
- If set it is false the op is pushed into the netty's buffer and non-writable status is bubbled up eventually forcing LedgerHandle to block and wait.
- If set to true it means that LedgerHandle knew that underlying bookie/channel is not writable, waited, given up, and is ok with failing this quickly. I.e. write op in this case waits for writable bookies in WQ, times out, submits op with the flag. Requests to non-writable bookies fail immediately triggering ensemble change as usual.

Reply:
My request is to shift the focus to the channel. All we need to know how many requests/data-size channel can accept once it turned non-writable. (another tunable)? Reset this counter as soon as channel turns writable. If the caller waited or not should not influence the channel to submit the request of fail-fast. That avoids sending flag down.